scripts/missing-ingest-data-fixer/ingest_query.py

```diff
@@ -0,0 +1,180 @@
+import json
+
+import requests as rq
+
+
+class IngestQuery:
+    TOKEN = '<REPLACE WITH FRESH TOKEN>'
+    BASE_URL = 'https://api.ingest.archive.data.humancellatlas.org'
+    DEV_BASE_URL = 'https://api.ingest.dev.archive.data.humancellatlas.org'
+    BASE_URL = DEV_BASE_URL  # for testing on dev
+
+    @staticmethod
+    def get_ingest_projects():
+        url = IngestQuery.BASE_URL + '/projects/search/catalogue'
+        projects = rq.get(url + "?size=1000", headers={'Authorization': IngestQuery.TOKEN}).json()
+        projects = projects['_embedded']['projects']
+
+        return projects
+
+    @staticmethod
+    def get_project_update_url_by_uuid(uuid):
+        url = IngestQuery.BASE_URL + f'/projects/search/findByUuid?uuid={uuid}'
+        response = rq.get(url)
+        if response.status_code == 200:
+            return response.json()['_links']['self']['href']
+
+        return None
+
+    @staticmethod
+    def patch_project(project_update_url, patch_data):
+        print(f'Updating project (project url: {project_update_url}) with {patch_data}')
+        response = rq.patch(project_update_url, data=json.dumps(patch_data), headers={'Authorization': IngestQuery.TOKEN, 'Content-Type': 'application/json'})
+        if response.status_code != 200:
+            print(f'Project with url:{project_update_url} failed with patching this payload: {patch_data}')
+
+    @staticmethod
+    def get_projects_from_dts_from_sheet(sheet_path):
+        with open(sheet_path, 'r') as f:
+            dts = f.read().splitlines()
+
+        dts = [row.split('\t') for row in dts]
+        headers = dts.pop(0)
+        list_projects = {}
+        for row in dts:
+            new_dict = {}
+            for i in range(len(headers)):
+                new_dict[headers[i]] = row[i]
+            list_projects[new_dict.get('UUID')] = new_dict
+
+        return list_projects
+
+    @staticmethod
+    def not_exists_by_doi(ingest_projects, dts_projects):
+        exists = False
+        not_in_ingest = {}
+        for project_dts in dts_projects:
+            for project_ingest in ingest_projects:
```

Comment:
this is too complex, i think you could have used a dictionary here to join the projects in dts and ingest

Reply:
This part of the utility class is not used for the missing organ. I can remove it if you want. I used it in some previous tasks. I just wanted to commit it to a repo to reuse it.